assets/src/application/ui/layout/entityList/EntityList/index.tsx

```diff
@@ -2,42 +2,33 @@ import React from 'react';
 import { __ } from '@wordpress/i18n';
 import { without } from 'ramda';
 import classNames from 'classnames';
+import { PaginationProps } from 'antd/lib/pagination';
 
 import { Entity } from '@edtrServices/apollo/types';
 import EmptyState from '@appDisplay/EmptyState';
+import { EntityPagination } from '../pagination';
 import './style.scss';
 
 interface EntityListProps {
 	entities: Entity[];
 	EntityGridView: React.ElementType;
 	EntityListView: React.ElementType;
-	htmlClass: string;
-	view: 'grid' | 'list';
-	noResultsText: string;
+	filterState?: any;
+	htmlClass?: string;
+	noResultsText?: string;
+	paginationProps: PaginationProps;
+	view?: 'grid' | 'list';
 }
 
-/**
- * EntityList
- * base component for displaying a list of entities (dates, tickets, etc)
- * as either a list table or grid of entity blocks
- *
- * @function
- * @param {Array} entities
- * @param {Component} EntityGridView
- * @param {Component} EntityListView
- * @param {string} htmlClass
- * @param {string} view
- * @param {string} noResultsText
- * @param {mixed} otherProps
- * @return {Component} list of rendered entities
- */
 const EntityList = ({
 	entities = [],
 	EntityGridView,
 	EntityListView,
+	filterState,
 	htmlClass = '',
-	view = 'grid',
 	noResultsText = '',
+	paginationProps,
+	view = 'grid',
```

Comment:
Why do we pass three props - `EntityGridView`, `EntityListView` and `view`? Why not just a generic prop like `component` and use that to render? The reason for that being the consumer (here dates `List`) is already aware which component to pass based on the current view.

Reply:
very good observation...
don't know if i have a good answer, but my guess is that it's because of how things were initially set up prior to things being abstracted, and that pattern got carried through multiple iterations without a second thought. Good example of how fresh eyes can see things that you have become blind too.

now that said, if we push all of the pagination and filtering responsibility into the EntityList, then this component will no longer know what the view is because that is not part of the dates filters but part of the generic entity filter bar.

Reply:
Made a small improvement here: https://github.com/eventespresso/event-espresso-core/pull/2431/commits/19d70d1299fb5a87efbae08cf82a77ed8089f63b